ttnn/cpp/ttnn/operations/eltwise/binary/binary_compiler_interface.hpp

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <exception>
+#include <functional>
+
+#include "third_party/json/json.hpp"
+#include "tt_metal/common/logger.hpp"
+#include "ttnn/compiler_interface/compiler_interface.hpp"
+#include "ttnn/graph/graph_processor.hpp"
+#include "ttnn/graph/graph_trace_utils.hpp"
+
+namespace ttnn::compiler_interface {
+
+template <auto BinaryFunction>
+QueryResponse binary_op_constraints(
+    Device* device, const TensorSpec& input_a, const TensorSpec& input_b, const TensorSpec& output) {
+    return op_constraints(device, [&]() {
+        const auto input_tensor_a = create_device_tensor(input_a, device);
+        const auto input_tensor_b = create_device_tensor(input_b, device);
```

Comment:
Input tensor creation may be dangerous here, you can ran into an out-of-memory error, even though an actual invocation of the op with pre-allocated inputs would succeed

Reply:
thanks for your feedback. do you think it is a real problem given it's guarded with NO_DISPATCH graph capture that doesn't really run allocator nor dispatches kernel?